target/linux/ath79/generic/base-files/etc/board.d/01_leds

```diff
@@ -407,6 +407,10 @@ yuncore,a770)
 	ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth1"
 	ucidef_set_led_switch "lan" "LAN" "green:lan" "switch0" "0x10"
 	;;
+xiaomi,aiot-ac2350)
+	ucidef_set_led_switch "wan_link" "WAN Link" "blue:wan" "switch0" "0x02" "0x0c"
+	ucidef_set_led_netdev "wan_data" "WAN Data" "orange:wan" "eth0.2" "tx rx"
```

Comment:
This lookds strange, is this vendor behavior? Apart from that, ucidef_set_led_switch is preferred.

Reply:
i'm responsible for this part of the code. the vendor behavior is that the "wan" led is actually an "internet" led that turns blue when there's a working internet connection, and orange otherwise (don't ask for the "not plugged in to wan port" behavior... i don't remember).

implementing something generic enough ended up being a nightmare, however. neither "switch" nor "netdev" seemed to do what i really wanted. so this was my compromise: solid blue light when WAN port connected to 100mbit or higher, and flashing orange for rx/tx. it doesn't look so bad in practice (since it's a single "led" on the case).

having said that, i'm open to the idea that this wasn't really the right solution :/

Reply:
Then please do what everybody else does, and what is somewhat OpenWrt default behavior, simply:

ucidef_set_led_switch "wan" "WAN" "blue:wan" "switch0" "0x2"

Reply:
The behavior of the vendor is that the "Internet" (WAN) indicator lights up orange while there is no connection to the servers on the Internet and lights up blue when there is a connection to external servers. The suggested code will only show the presence of a link on WAN. That's probably enough to start with.

Reply:
The suggestion was applied in d332184